### DISCUSSION TITLE
CodeChecker on Android

### DISCUSSION THREAD

**Discussion by nikey02** [Ericsson/codechecker#3304] [General]
hello everyone 
i am trying to build codechecker on android under [termux](https://github.com/termux/)
i install all Requirements to build codechecker from source 
but stuck at

![IMG_20210506_175219](https://user-images.githubusercontent.com/68458326/117298231-c1845480-ae94-11eb-82e3-bf01f8e57efe.jpg)

thanks

**Comment by csordasmarton** [comment]
@nikey02 The source code of the logger contained a couple of compiler warnings and your system converted it to compiler error. I created a patch which will solve these compiler warnings: #3305.

Unfortunately the first error message is not a compiler error but your compiler complains about the `wordexp.h` header file, that it is missing: https://man7.org/linux/man-pages/man3/wordexp.3.html.

**Reply by nikey02** [reply]
@csordasmarton  thanks for swift reply 
after searching around i found out that 
wordexp.h is not available on android but 
found solution 
i will update the patch give another try

if it works it will be great to have codechecker on android

**Reply by nikey02** [reply]
i tried again with some hacks 
i have to switch to gcc for those errors because default compiler is clang 
clang-12 was giving those errors but manage to build ldlogger.so
but there are several road blocks for Android now while installing vue-cli npm fails at chromedriver which is not available for Android arm64 obviously : (

again thanks  @csordasmarton  for quick fix 
i will try again later
if someone manage to build on Android or have workarounds please drop the comment

**Reply by csordasmarton** [reply]
@nikey02 You can remove this dependency from the list, because this dependency is only for running GUI tests:
https://github.com/Ericsson/codechecker/blob/46cca7e55d1f6dd3a726fb17794917036948efad/web/server/vue-cli/package.json#L56
So this dependency is not needed to build the package.

Also by default the make package will build the UI code if it's not built yet or the UI code is changed. If you wouldn't like to build the UI code you can set the `BUILD_UI_DIST` environment variable to `NO` before the package build: `BUILD_UI_DIST=NO make package`. It is described here: https://github.com/Ericsson/codechecker#linux in the notes section.

Also I don't no whether docker is available on Android, but if yes, we have pre-built docker images for the web server part what you can use to start a web server on your host: https://hub.docker.com/r/codechecker/codechecker-web

**Reply by nikey02** [reply]
@csordasmarton  great i will give another shot . also i have to check other dependencies .
docker can be run on android but it needs 
rooting and patching kernel

**Reply by csordasmarton** [reply]
@nikey02 From the 6.16.0 official pypi packages are also available and I think it can be used on your system too if pip command is available. You can give it a try: https://pypi.org/project/codechecker/
